packages/react/src/components/searchInput/SearchInput.tsx

```diff
@@ -133,43 +156,40 @@ export const SearchInput = <SuggestionItem,>({
     reset,
   } = useCombobox<SuggestionItem>({
     items: suggestions,
-    onInputValueChange: (e) => {
-      if (isControlledComponent) {
-        onChange(e.inputValue);
-      } else {
-        setInternalValue(e.inputValue);
-      }
-    },
     onStateChange(props) {
       const { ItemClick } = useCombobox.stateChangeTypes;
       if (props.type === ItemClick) {
```

Comment:
By testing this locally, I detected that only having values submitted via ItemClick removes some other functionalities when SearchInput is used as a controlled component (check story WithDefaultValue). At least resetting value or selecting the value from the list with keyboard. Luckily the docs have [stateChangeTypes](https://github.com/downshift-js/downshift/tree/master/src/hooks/useCombobox#statechangetypes) that can help cover this! Check them out if there's something else that might help, there is stuff about Home and End keys as well.

Here's a suggestion to support at least resetting and keyboard selection, feel free to add any more that you find useful:
`const { InputKeyDownEnter, ItemClick, FunctionReset } = useCombobox.stateChangeTypes;
if (props.type === ItemClick || props.type === FunctionReset || props.type === InputKeyDownEnter) {
        if (props.type !== FunctionReset) isItemClicked.current = true;
        dispatchValueChange(props.inputValue);
        submitValue(props.inputValue);
 }`

Reply:
Oh, what a bug! The reset check is even in the tests. But it tested uncontrolled component which worked.

The state change tracks now also `FunctionReset`  and `InputKeyDownEnter`. This change caused more side-effects and changes to the component. There were double submits.

I added a more tests so these won't occur again undetected.

Home and End keys must be trapped onKeyDown to prevent Downshift default behaviour.

Reply:
Nice! Now those are covered. But unfortunately I found another bug and console.error.. 🙈 Bit of edge-cases though but still.

1. `SearchInput.tsx:245 Uncaught TypeError: Cannot read properties of undefined (reading 'length')` crash occurs when you type a value which is suggested -> click it -> enter text into input and remove it -> set focus elsewhere on page
2. related this is a console error about switching to controlled -> uncontrolled component. If you do the same steps with WithDefaultValue story there's a console.error about this

Reply:
Wow... I couldn't quite follow your steps, but found easier way

If you type **exactly** the same value that is suggested, like "Apple", and click the suggestion, it crashes. The object passed to "onStateChange" has no inputValue-prop when it matches the suggested value.

Reply:
Fixed now. The behaviour now slightly differs from the current version in master.
The value is now submitted when the value in the input matches the value of the selected item in the dropdown. 

But this is actually  a fix, because value is submitted in all other cases when item is selected. For example if "Appl" is written and "Apple" is selected. :D 

Added more tests.

Reply:
Nicely fixed! I'm not finding anymore bugs in functionality. This whole PR is a Major improvement to this component! 🥇 Has @minevala anything to add to the latest commits?

Reply:
I marked this approved for now as it seems that the tests cover the prior issues nicely and they seem to be passing :)